File: Yarışma Cevapları/Ceylin ÇALTEPE - Ramazan YILDIRIM - Ayşe Nur İBİŞ - Nimet OKUR/Soru6.java
public class Soru6 {
    public static void main(String[] args) {
        int sayac = 0, carpan, mukemmelSayi;
        for (int sayi = 2; sayi <= 500; sayi++) {
            int kontrol = 0;
            carpan = (int) (Math.pow(2, sayi) - 1);
            for (int i = 2; i < sayi / 2; i++) {
                if ((sayi % i == 0)) {
                    kontrol = 1;
                    break;
                }
            }
            if (kontrol == 0) {
                sayac++;
                mukemmelSayi = carpan * (int) Math.pow(2, sayi - 1);
                System.out.println(mukemmelSayi);

            }

        }


    }
}

File: Yarışma Cevapları/Berk Sedat KABADAYI - Ali İHSAN/Soru 6.js
let arr = [];

function prime(n) {
    for (let a=2 ; a < 500; a++) {
        if (n%a == 0 && a!=n) return false;
        if (a == 499) return true;
    }
}

function perfectNum(p) {

    if (prime(Math.pow(2,p) - 1) && prime(p)) {
        let n = Math.pow(2,p-1) * (Math.pow(2,p) - 1)
        if (n < 500)  {
            return n;
        } else return false;
    } else return false;
}

for (let b = 2; b < 500; b++) {
    if (prime(b) && perfectNum(b)) arr.push(perfectNum(b))
}

console.log(arr);